lib/manageiq/providers/ibm_cloud/cloud_tools/common/proxy.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module ManageIQ
+  module Providers
+    module IbmCloud
+      module CloudTools
+        module Common
+          # Set proxy configurations dynamically.
+          class Proxy
+            def initialize
+              @keys = %i[address port username password headers]
+              @keys.each { |key| self.class.attr_accessor(key) }
```

Comment:
This can be done right in the Proxy class itself as opposed to initialize, which will save on doing this once per instance, which can be expensive.

```ruby
class Proxy
  attr_accessor :address, :port, :username, :password, :headers]
```

Reply:
@keys is used in to_hash which is used to pass the proxy data to the IBM SDK init. I could use the standard attr_accessor, but then I'd have to maintain two lists. Also there should only be 1 instance of Proxy created per instance of CloudTool, so the performance hit shouldn't be too significant.

Reply:
It's still preferable to do it just once at the class level as this is essentially static data.

```ruby
class Proxy
  ACCESSOR_KEYS = %i[address port username password headers].freeze
  attr_accessor *ACCESSOR_KEYS

#...

  def to_hash
    self.class::ACCESSOR_KEYS.each_with_object({}) { |k, h| h[k] = instance_variable_get("@#{key}") }.compact
  end
```

Reply:
Reimplemented suggested code with minor fixes.